### PR TITLE
Move util classes to java 8

### PR DESCRIPTION
**Pull request by kageiit**
Ref #329 
Compared the generated buck files and ensured there are no differences other than the ones intended

### CODE REVIEW

File: buildSrc/src/main/java/com/uber/okbuck/core/util/ReplaceUtil.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            OutputStream outputStream = new FileOutputStream(destination);
+            IOUtils.copy(replacingReader, outputStream, Charset.defaultCharset());
+            IOUtils.closeQuietly(outputStream);
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private interface TemplateResolver {
+
```
Comment (raviagarwal7) [inline buildSrc/src/main/java/com/uber/okbuck/core/util/ReplaceUtil.java]: remove new line

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -10,6 +10,12 @@ repositories {
     jcenter()
 }
 
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
+sourceSets.main.java.srcDirs = []
```
Comment (raviagarwal7) [inline buildSrc/build.gradle]: sourceSets.main.java.srcDirs = ["src/main/java"]?

File: buildSrc/src/main/java/com/uber/okbuck/core/util/LintUtil.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    private static final String LINT_DEPS_CONFIG = OkBuckGradlePlugin.BUCK_LINT + "_deps";
+    private static final String LINT_DEPS_CACHE = OkBuckGradlePlugin.DEFAULT_CACHE_PATH + "/lint";
+    private static final String LINT_VERSION_FILE = LINT_DEPS_CACHE + "/.lintVersion";
+    private static final String LINT_DEPS_BUCK_FILE = "lint/BUCK_FILE";
+
+    public static final String LINT_DEPS_RULE = "//" + LINT_DEPS_CACHE + ":okbuck_lint";
+
+    private LintUtil() {}
+
+    public static String getDefaultLintVersion(Project project) {
```
Comment (justinmuller) [inline buildSrc/src/main/java/com/uber/okbuck/core/util/LintUtil.java]: `@Nullable`

File: buildSrc/src/main/java/com/uber/okbuck/core/util/ProjectUtil.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.plugins.PluginContainer;
+
+import java.io.File;
+import java.util.Map;
+
+public final class ProjectUtil {
+
+    private ProjectUtil() {
+        // no instance
```
Comment (justinmuller) [inline buildSrc/src/main/java/com/uber/okbuck/core/util/ProjectUtil.java]: Other util classes don't comment the constructor, just have `{}`

File: buildSrc/src/main/java/com/uber/okbuck/core/util/ReplaceUtil.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        @Override
+        public String resolveTemplate(String template) {
+            return this.templateMap.get(template);
+        }
+    }
+
+    private static final class TemplateReader extends Reader {
+
+        PushbackReader pushbackReader = null;
```
Comment (justinmuller) [inline buildSrc/src/main/java/com/uber/okbuck/core/util/ReplaceUtil.java]: Unnecessary initialization?

File: buildSrc/src/main/java/com/uber/okbuck/core/util/ReplaceUtil.java [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                    .resolveTemplate(this.templateBuffer.toString());
+
+            if (this.template == null) {
+                this.template = "${" + this.templateBuffer.toString() + "}";
+            }
+            if (this.template.length() == 0) {
+                return read();
+            }
+            return this.template.charAt(this.templateIndex++);
+
```
Comment (justinmuller) [inline buildSrc/src/main/java/com/uber/okbuck/core/util/ReplaceUtil.java]: Unnecessary newlines.

File: buildSrc/src/main/java/com/uber/okbuck/core/util/ReplaceUtil.java [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+        }
+
+        @Override
+        public int read(CharBuffer target) throws IOException {
+            throw new RuntimeException("Operation Not Supported");
+        }
+
+        @Override
+        public int read() throws IOException {
+            if (this.template != null) {
```
Comment (justinmuller) [inline buildSrc/src/main/java/com/uber/okbuck/core/util/ReplaceUtil.java]: Is the use of `this.` consistent with the rest of the codebase?

File: buildSrc/src/main/java/com/uber/okbuck/core/util/RetrolambdaUtil.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+import com.uber.okbuck.core.model.base.Scope;
+import com.uber.okbuck.extension.RetrolambdaExtension;
+
+import org.apache.commons.lang.StringUtils;
+import org.gradle.api.Project;
+
+import java.io.File;
+import java.util.Collections;
+
+class RetrolambdaUtil {
```
Comment (justinmuller) [inline buildSrc/src/main/java/com/uber/okbuck/core/util/RetrolambdaUtil.java]: `final`?

File: buildSrc/src/main/java/com/uber/okbuck/core/util/RetrolambdaUtil.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+    private static final String RETROLAMBDA_CMD_TEMPLATE =
+            "(read CLASSES_DIR && java -Dretrolambda.inputDir=\"${CLASSES_DIR}\" " +
+                    "-Dretrolambda.classpath=\"${COMPILATION_BOOTCLASSPATH}\":"
+                    + "\"${COMPILATION_CLASSPATH}\":"
+                    + "\"${CLASSES_DIR}\"";
+
+    private RetrolambdaUtil() {}
+
+    public static String getRtStubJarRule() {
```
Comment (justinmuller) [inline buildSrc/src/main/java/com/uber/okbuck/core/util/RetrolambdaUtil.java]: `public` methods in a package-private util class?

File: gradle/wrapper/gradle-wrapper.properties [hunk 10]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (justinmuller) [inline gradle/wrapper/gradle-wrapper.properties]: Intentional?

File: buildSrc/build.gradle (hunk 2 above)
Comment (kageiit) [inline buildSrc/build.gradle]: Currently we need joint compilation of both groovy and java so they need to be in the same sourceset. As more classes move over to java, we will remove the groovy sourceset entirely

**Pull request merged**